processing/src/main/java/org/apache/druid/query/operator/OffsetLimit.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.operator;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import org.apache.druid.error.DruidException;
+import java.util.Objects;
+
+public class OffsetLimit
```

Comment:
Also can we call this a OffsetLimitOperator ?

Reply:
yes; we should consolidate the 2 classes - I was not able to use it and move it right away; it would have been a refactor step which I wanted to avoid as it would touch some places of the planner as well - so I would rather put it into a separate PR; in which only that's happening.

Since we have an `Operator` interface; I think it might be a bit confusing to add that keyword to this class's name - I think after we will only have one it might be already better.

Reply:
Since this is a release blocker, let's not try to refactor everything here and fix as a followup